netbench-orchestrator/README.md

```diff
@@ -0,0 +1,103 @@
+
+## Goals
+Often developers of transport protocols are interested in gathering performance data for the protocol
+they are developing. Netbench is a tool which can be used to measure this performance data.
+However, its often necessary to run Netbench scenarios in the cloud so that the results better match
+production systems. The goal of this project is to automate Netbench runs in the cloud.
+
+## Getting started
+
+**Pre-requisites**
+- Built and include [netbench](https://github.com/aws/s2n-netbench) utilities (`cargo build`)
+  - Include in PATH `export PATH="s2n-netbench/target/release/:$PATH"`. Test with `which s2n-netbench`
+- AWS cli is installed. Test with `which aws`
+- An AWS account with some infrastructure configured. TODO: provide an easy way to do this
+  - Make sure AWS credentials are included in your shell environment
+- The ec2 SSH key name is correctly set in state.rs (make this configurable)
+
+**Running**
+
+```
+git clone git@github.com:aws/s2n-netbench.git && cd s2n-netbench
+
+# Run the orchestrator
+make run_orchestrator
+```
+
+## Project Overview
+Since the goal of the Orchestrator is to run workloads on remote servers, its best to think
+of the project as two components; stuff that runs locally vs remotely.
+
+- The **Orchestrator** runs **locally**, and is responsible for spinning up servers, and
+configuring them to run netbench. The orchestrator creates Russula 'Coordinators' to
+coordinate the remote netbench runs (done over TCP connection).
+- **SSM** and **Russula** run **remotely** on the hosts. SSM is used primarily for async
+tasks such as installing/updating dependencies and uploading netbench data to s3. SSM is
+also used to start the Russula worker process on each host. Russula is primarily used for
+timing sensitive tasks such netbench, which require starting and stopping multiple servers
+and clients processes across multiple hosts. Specifically, the 'Worker' component of a
+protocol is executed on the remote host, while the 'Coordinator' component is run locally
+as part of the Orchestrator.
+
+### Debugging
+As discussed in the above overview, there are processes that run locally and those that run
+remotely. This sections describes how to go about debugging each component.
+
+#### Local
+**Orchestrator**
+The Orchestrator is Rust code and ships with [tracing](https://docs.rs/tracing/latest/tracing/)
+support. Logs are written to a file `orch_proj/target/russula.log*` file on the host. The
+`make run_orchestrator` command enables sane log levels via `RUST_LOG=...` but these can be
+changed as desired.
+
+#### Remote
+**SSH access**
+ec2 accepts the name of a ssh-key when creating a new host. This is set to a default value
+under the value `ssh_key_name` in the [state.rs](/src/state.rs) file. By providing this key
+it is possible to ssh onto the remote host locally: `ssh -oStrictHostKeyChecking=no ec2-user@x.x.x.x`.
+Its also possible to ssh onto a host from the ec2 console on AWS.
```

Comment:
We will specifically be launching al2023 so ec2-user should be accurate. The orchestrator relies on ssm so it will prob take some effort to run on other platforms. https://github.com/aws/s2n-netbench/blob/ak-fullOrchImport/netbench-orchestrator/src/orchestrator/state.rs#L28